demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -105,68 +106,15 @@ int main(int argc, char *argv[]) {
         Core ie;
 
-        slog::info << "Device info: " << slog::endl;
-        std::cout << ie.GetVersions(FLAGS_d);
-
-        /** Load extensions for the plugin **/
-        if (!FLAGS_l.empty()) {
-            // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
-            IExtensionPtr extension_ptr = make_so_pointer<IExtension>(FLAGS_l.c_str());
-            ie.AddExtension(extension_ptr, "CPU");
-        }
-        if (!FLAGS_c.empty()) {
-            // clDNN Extensions are loaded from an .xml description and OpenCL kernel files
-            ie.SetConfig({{PluginConfigParams::KEY_CONFIG_FILE, FLAGS_c}}, "GPU");
-        }
-
-        /** Per layer metrics **/
-        if (FLAGS_pc) {
-            ie.SetConfig({ { PluginConfigParams::KEY_PERF_COUNT, PluginConfigParams::YES } });
-        }
-
-        std::map<std::string, std::string> userSpecifiedConfig;
-        std::map<std::string, std::string> minLatencyConfig;
-
-        std::set<std::string> devices;
-        for (const std::string& device : parseDevices(FLAGS_d)) {
-            devices.insert(device);
-        }
-        std::map<std::string, unsigned> deviceNstreams = parseValuePerDevice(devices, FLAGS_nstreams);
-        for (auto& device : devices) {
-            if (device == "CPU") {  // CPU supports a few special performance-oriented keys
-                // limit threading for CPU portion of inference
-                if (FLAGS_nthreads != 0)
-                    userSpecifiedConfig.insert({ CONFIG_KEY(CPU_THREADS_NUM), std::to_string(FLAGS_nthreads) });
-
-                if (FLAGS_d.find("MULTI") != std::string::npos
-                    && devices.find("GPU") != devices.end()) {
-                    userSpecifiedConfig.insert({ CONFIG_KEY(CPU_BIND_THREAD), CONFIG_VALUE(NO) });
-                } else {
-                    // pin threads for CPU portion of inference
-                    userSpecifiedConfig.insert({ CONFIG_KEY(CPU_BIND_THREAD), CONFIG_VALUE(YES) });
-                }
-
-                // for CPU execution, more throughput-oriented execution via streams
-                userSpecifiedConfig.insert({ CONFIG_KEY(CPU_THROUGHPUT_STREAMS),
-                                (deviceNstreams.count(device) > 0 ? std::to_string(deviceNstreams.at(device))
-                                                                  : CONFIG_VALUE(CPU_THROUGHPUT_AUTO)) });
-
-                minLatencyConfig.insert({ CONFIG_KEY(CPU_THROUGHPUT_STREAMS), "1" });
-            } else if (device == "GPU") {
-                userSpecifiedConfig.insert({ CONFIG_KEY(GPU_THROUGHPUT_STREAMS),
-                                (deviceNstreams.count(device) > 0 ? std::to_string(deviceNstreams.at(device))
-                                                                  : CONFIG_VALUE(GPU_THROUGHPUT_AUTO)) });
+        std::string deviceInfo;
+        std::string deviceString = FLAGS_d;
+        formatDeviceString(deviceString);
```

Comment:
You will need to copy the string for both demos. I suggest returning a copy from `formatDeviceString()`.

Reply:
I also use `deviceString` in `ie.LoadNetwork()`, so I think It's better to do the formatting task only once and then just make string copies when needed.

Reply:
`deviceString` is not modified before `ie.LoadNetwork()`, you would reuse what `formatDeviceString()` returned.